File: mytasks-app-backend/Models/Task.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace mytasks.Models {
    public class Task {

        [Key]
        public int TaskId { get; set; }
        public string TaskName { get; set; }

        public DateTime Created { get; set; }

        public DateTime? Closed { get; set; }

        public string Description { get; set; }

        public decimal PlannedHours { get; set; }

        public decimal ActualHours { get; set; }

        public TaskState State { get; set; }

    }
}

File: mytasks-app/src/app/services/tasks/task.ts
import { TaskState } from "./task-state";

export class Task {
    taskId: number;

    taskName: string;

    state: TaskState;

    created: Date;

    closed: Date;

    description: string;

    plannedHours: number;

    actualHours: number;

    projectId: number;
}

export interface TaskView {
    taskId: number;

    taskName: string;

    state: TaskState;

    created: Date;

    plannedHours: number;

    actualHours: number;

    projectName: number;
}